child/child.go

```diff
@@ -362,9 +362,13 @@ func (c *Child) kill() {
 	exited := false
 	process := c.cmd.Process
 
-	select {
-	case <-c.stopCh:
-	case <-c.randomSplay():
+	if c.cmd.ProcessState == nil {
```

Comment:
Just noting that this should be right but there is also an `Exited()` bool -- seems to be something that would make sense here but if the nil check works then 👍 https://godoc.org/os#ProcessState.Exited.